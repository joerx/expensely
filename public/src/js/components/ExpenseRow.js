import React from 'react';
import ucfirst from '../helpers/ucfirst';
import printf from 'printf';

// Single expense
// - onDelete()
const ExpenseRow = ({label, amount, actionIcon, onActionClick}) => (
  <div className="expense-row">
    {ucfirst(label)}
    <span className="pull-right">{printf('%.2f S$', amount)}
      <a onClick={onActionClick}>
        <span className={`glyphicon glyphicon-${actionIcon}`}/>
      </a>
    </span>
  </div>
);

export default ExpenseRow;
