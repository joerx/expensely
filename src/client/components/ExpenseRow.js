import React from 'react';
import ucfirst from '../helpers/ucfirst';
import printf from 'printf';

// Single expense
// - onDelete()
const ExpenseRow = ({label, amount, actionIcon, onActionClick}) => {
  return (<div className="expense-row">
    <span className="expense-label">{ucfirst(label)}</span>
    <span className="pull-right expense-amount">{printf('%.2f S$', amount)}
      <a className="row-action" onClick={onActionClick}>
        <span className={`glyphicon glyphicon-${actionIcon}`}/>
      </a>
    </span>
  </div>)
};

export default ExpenseRow;
