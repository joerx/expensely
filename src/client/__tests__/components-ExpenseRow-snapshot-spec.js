import React from 'react';
import ExpenseRow from '../components/ExpenseRow';
import renderer from 'react-test-renderer';

function setup() {
  const props = {
    actionIcon: 'ok',
    onActionClick: jest.fn(),
    label: 'some label text',
    amount: 20
  }
  const wrapper = shallow(<ExpenseRow {...props}/>);
  return {props, wrapper};
}

describe('ExpenseRow component', () => {

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should render itself', () => {
    const props = {
      actionIcon: 'ok',
      onActionClick: jest.fn(),
      label: 'some label text',
      amount: 20
    };
    const component = renderer.create(
      <ExpenseRow {...props}/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
