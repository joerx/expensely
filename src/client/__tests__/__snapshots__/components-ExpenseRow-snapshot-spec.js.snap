exports[`ExpenseRow component should render itself 1`] = `
<div
  className="expense-row">
  <span
    className="expense-label">
    Some label text
  </span>
  <span
    className="pull-right expense-amount">
    20.00 S$
    <a
      className="row-action"
      onClick={[Function]}>
      <span
        className="glyphicon glyphicon-ok" />
    </a>
  </span>
</div>
`;
